### DISCUSSION TITLE
Embed a youtube video in a privacy friendly way

### DISCUSSION THREAD

**Discussion by Andre601** [mkdocs/mkdocs#3874] [Q&A]
I would like to embed a youtube video in a somewhat privacy friendly way, preferably not having any info sent to youtube through the embedded player.

From what I could gather is the default iframe youtube offers not a solution for this, as they would collect info through that, no matter what.
I assume I could use some form of proxy available to have it go through that and avoid data collection?

Any recommendations here for how to properly embed videos in a privacy safe manner?

**Comment by markallasread** [comment]
I think the best option is to embed the videos using youtube-nocookie.com vs. youtube.com — it's youtube's enhanced privacy service that'll ensure no data is sent to youtube until a user plays the video: Caveat, the user is always going to send traffic to youtube when playing the video, it's the nature of the internet.

Download the video / host it on your site if you really wish to avoid sending data to a 3rd party service.

**Reply by atraining** [reply]
Even the ticket is old, so just FYI: @Andre601 I did create a hook: https://github.com/atraining/idp-software.com/blob/main/hooks/youtube_hook.py
